services/management/dbaas/pxc_cluster_service.go

```diff
@@ -185,6 +203,112 @@ func (s PXCClustersService) CreatePXCCluster(ctx context.Context, req *dbaasv1be
 	return &dbaasv1beta1.CreatePXCClusterResponse{}, nil
 }
 
+func (s PXCClustersService) fixCreateRequestWithDefaults(ctx context.Context, req *dbaasv1beta1.CreatePXCClusterRequest) error {
+	if req.Name != "" {
+		r, _ := regexp.Compile("^[a-z]([-a-z0-9]*[a-z0-9])?$")
+		if !r.MatchString(req.Name) {
+			return errInvalidClusterName
+		}
+	}
+	if req.Params == nil {
+		req.Params = &dbaasv1beta1.PXCClusterParams{}
+	}
+
+	if req.Params.ClusterSize < 1 {
+		req.Params.ClusterSize = pxcDefaultClusterSize
+	}
+
+	if req.Params.Pxc == nil {
+		req.Params.Pxc = &dbaasv1beta1.PXCClusterParams_PXC{}
+	}
+
+	if req.Params.Pxc.DiskSize == 0 {
+		req.Params.Pxc.DiskSize = pxcDefaultDiskSize
+	}
+
+	if req.Params.Pxc.ComputeResources == nil {
+		req.Params.Pxc.ComputeResources = &dbaasv1beta1.ComputeResources{
+			CpuM:        pxcDefaultCpuM,
+			MemoryBytes: pxcDefaultMemoryBytes,
+		}
+	}
+	if req.Params.Pxc.ComputeResources.CpuM == 0 {
+		req.Params.Pxc.ComputeResources.CpuM = pxcDefaultCpuM
+	}
+	if req.Params.Pxc.ComputeResources.MemoryBytes == 0 {
+		req.Params.Pxc.ComputeResources.MemoryBytes = pxcDefaultMemoryBytes
+	}
+
+	if !req.Expose {
+		// clean up just in case the request has HAProxy or ProxySql but enabled is false.
+		// This was we can skip setting the defaults in the ifs below.
+		req.Params.Haproxy = nil
+		req.Params.Proxysql = nil
+	}
```

Comment:
I'm not sure that we should clean values for Haproxy and Proxysql, because expose stands for making DB cluster available outside of kubernetes cluster, but HAProxy or ProxySQL still required for internal usage.
So expose and Proxysql/Haproxy is not related to each other.